### ISSUE TITLE
Crashing preferences

### ISSUE THREAD

**Issue by hellobrother222** [jacobcxdev/FBSpNOsor#6]
The preferences are crashing, I can't configure the tweak. It still works though, I'm not seeing any ads. Running odysseyra1n on iOS 13.6.1

**Comment by jacobcxdev** [comment]
Do you have a crash log?

**Comment by hellobrother222** [comment]
https://pastebin.com/fUuksDHG

**Comment by jacobcxdev** [comment]
Possible duplicate of #5? Please install libhooker Configurator and try disabling every tweak besides FBSpNOsor, as it may be an incompatibility with another tweak.

**Comment by hellobrother222** [comment]
It still crashes even when everything is disabled except FBSpNosor and PreferenceLoader

**Comment by jacobcxdev** [comment]
Huh, that's really strange — I don't see why it would crash there at all. The settings page only offers an enable/disable switch, so if the tweak works then it's not _really_ an issue. That being said, I will investigate why it's crashing if I have the time.

**Comment by hellobrother222** [comment]
All good man, I was able to enable/disable the tweak by editing the preferences file through filza. Keep up the good work

**Comment by megatech1966** [comment]
It's crashing for me to. I'm also having a problem with Facebook videos. They sometimes crash.

**Comment by megatech1966** [comment]
I have found something out. If you close Facebook, go to fbspnosor in settings. It will crash. Do a respring and it works again. Will test to see Facebook videos still crash before I close Facebook. 
iPhone X. iOS 14.3 Taurine. Installed fbspnosor with Cydia.

**Comment by jacobcxdev** [comment]
> I have found something out. If you close Facebook, go to fbspnosor in settings. It will crash. Do a respring and it works again. Will test to see Facebook videos still crash before I close Facebook.
> iPhone X. iOS 14.3 Taurine. Installed fbspnosor with Cydia.

Interesting. Are you able to provide a crash log too?

**Comment by megatech1966** [comment]
Log file

https://www.dropbox.com/s/7qh5vxz5z94pdhv/File%205-5-21%2C%208%2057%2041%20pm.txt?dl=0

**Comment by jacobcxdev** [comment]
Thanks. I, uh, have literally no idea wtf is going on in that crash log to mean it's crashing in dyld. I'll ask around and see if anyone else knows 😕.

**Comment by megatech1966** [comment]
Facebook videos are still crashing. Seems to be a different reason. Here is the log file. 

https://www.dropbox.com/s/5tn1xaosua9waym/File%206-5-21%2C%2012%2017%2012%20am.txt?dl=0

**Comment by jacobcxdev** [comment]
> Facebook videos are still crashing. Seems to be a different reason. Here is the log file. 
> 
> https://www.dropbox.com/s/5tn1xaosua9waym/File%206-5-21%2C%2012%2017%2012%20am.txt?dl=0

I don't appear anywhere in that particular crash log; perhaps it's a bug in FB or with a different tweak? Try disabling all other tweaks via iCleaner Pro, libhooker Configurator, etc. and let me know if it still occurs.

**Comment by megatech1966** [comment]
Tried that. Still crashes. Am trying to see if it crashes with fbspnosor uninstalled. So far. No crashing.

**Comment by jacobcxdev** [comment]
Hmm, strange. I'll see if I can reproduce it when I have time.

**Comment by megatech1966** [comment]
It seems to have a problem with taurine. I have tested it on my iPad Air 2 iOS 14.4 checkrain. Works fine. Videos don't crash.

**Comment by megatech1966** [comment]
Tried another Facebook ad blocker. Videos crash with that one as well. iPhone X iOS 14.3 Taurine

**Comment by megatech1966** [comment]
I'm pretty sure it crashes when it tries to play a ad while watching a video.

**Comment by megatech1966** [comment]
It has a problem with the Taurine jailbreak. Checkrain works fine. It crashes when it tries to play a ad while watching a video.

**Comment by jacobcxdev** [comment]
Thanks, I'll pass this onto CoolStar.

**Comment by megatech1966** [comment]
Version 1.0 of another Facebook ad blocker seems to work fine.

**Comment by jacobcxdev** [comment]
> Version 1.0 of another Facebook ad blocker seems to work fine. 

Is it open source?

**Comment by megatech1966** [comment]
https://github.com/haoict/facebook-no-ads

**Comment by jacobcxdev** [comment]
It strikes me as strange that @haoict's tweak works fine when it supposedly uses my code. It's probably a weird issue with Taurine tbh.

**Comment by megatech1966** [comment]
Yes and something he changed in his updates and made it crash with Taurine. I can test all of his versions to see which of them work, if you like?

**Comment by megatech1966** [comment]
I tested all the versions. Only version 1.0 works with Taurine.
